src/Parser.php

```diff
@@ -3,14 +3,12 @@
 
 namespace Lcobucci\JWT;
 
-use InvalidArgumentException;
-
 interface Parser
 {
     /**
      * Parses the JWT and returns a token
      *
-     * @throws InvalidArgumentException
+     * @throws InvalidArgument
```

Comment:
We have a few scenarios where `InvalidArgumentException` was being thrown:

1. Strings with incorrect basic JWT structure
1. Incorrect type for headers/claims
1. Unsupported encrypted tokens

I reckon the first two could be grouped but the third is quite a different thing. Maybe we should have 2 different exceptions here...

Reply:
Don't have proper suggestions atm, though (shaaame)